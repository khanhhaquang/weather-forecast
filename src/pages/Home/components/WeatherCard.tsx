import { META_WEATHER_DOMAIN } from 'config';
import styled from 'styled-components';
import { getWeekDay, roundVal } from '../helpers';

const StyledWrapper = styled.div`
  display: flex;
  flex-direction: column;
  background-color: lightblue;
  border-radius: 4px;
  padding: 8px;
`;

const DateText = styled.h2`
  font-size: 16px;
  line-height: 24px;
  font-weight: 600;
  color: #337ab7;
`;

const StateWrapper = styled.div`
  display: flex;
  margin-bottom: 16px;
  font-size: 12px;
  line-height: 18px;

  span {
    margin-left: 4px;
  }
`;

type TProps = {
  minTemp: number;
  maxTemp: number;
  day: Date;
  stateImg: string;
  stateName: string;
};

const WeatherCard: React.FC<TProps> = ({
  minTemp,
  maxTemp,
  day,
  stateImg,
  stateName,
  ...rest
}: TProps) => (
  <StyledWrapper {...rest}>
    <DateText>{getWeekDay(new Date(day))}</DateText>
    <StateWrapper>
      <img
        alt="state"
        width={24}
        height={24}
        src={`${META_WEATHER_DOMAIN}/static/img/weather/png/64/${`${stateImg}.png`}`}
      />
      <span>{stateName}</span>
    </StateWrapper>
    <span>Max: {roundVal(maxTemp)}°C</span>
    <span>Min: {roundVal(minTemp)}°C</span>
  </StyledWrapper>
);
export default WeatherCard;
