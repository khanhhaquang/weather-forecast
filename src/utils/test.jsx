/* eslint-disable import/no-unresolved */
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { render as rtlRender } from '@testing-library/react';
import rootReducer from 'pages/reducer';

function render(ui, { initialState, store = createStore(rootReducer, initialState) } = {}) {
  return {
    ...rtlRender(<Provider store={store}>{ui}</Provider>),
    // adding `store` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    store,
  };
}

export * from '@testing-library/react';
export { render };
