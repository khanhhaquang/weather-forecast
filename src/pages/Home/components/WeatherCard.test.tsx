import { render, screen } from 'utils/test';
import WeatherCard from './WeatherCard';

describe('Home/WeatherCard', () => {
  const mockState = {
    home: {},
  };

  const mockProps = {
    minTemp: '2',
    maxTemp: '3',
    day: '1/1/1999',
    stateImg: 'abc',
    stateName: 'xyz',
  };

  it('should render correctly', () => {
    const { asFragment } = render(<WeatherCard {...mockProps} />, {
      initialState: { ...mockState },
    });
    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByText(/Max/i)).toBeInTheDocument();
    expect(screen.getByText(/Min/i)).toBeInTheDocument();
  });
});
