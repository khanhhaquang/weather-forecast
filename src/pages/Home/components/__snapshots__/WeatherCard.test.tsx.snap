// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home/WeatherCard should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnxRM gCZTqE"
  >
    <h2
      class="sc-gtsrHT ebtAAb"
    >
      Friday
    </h2>
    <div
      class="sc-dlnjwi QbBjM"
    >
      <img
        alt="state"
        height="24"
        src="https://meta-weather.now.sh/static/img/weather/png/64/abc.png"
        width="24"
      />
      <span>
        xyz
      </span>
    </div>
    <span>
      Max: 3°C
    </span>
    <span>
      Min: 2°C
    </span>
  </div>
</DocumentFragment>
`;
